packages/controller-config/src/types.ts

```diff
@@ -26,8 +26,12 @@ export const controllerConfigSchema = yup
       .oneOf(["gcp", "aws"])
       .required("must specify a target (gcp | aws)"),
     region: yup.string().required("must specify region"),
-    logRetention: yup.number().required("must specify log retention"),
-    metricRetention: yup.number().required("must specify metric retention"),
+    logRetentionDays: yup
+      .number()
+      .required("must specify log retention in number of days"),
+    metricRetentionDays: yup
+      .number()
+      .required("must specify metric retention in number of days"),
```

Comment:
Another change that should be a no-brainer on the one hand, but that will lead to funny upgrade effects when mixing old controller config with new controller version.

Reply:
We should, at some point, move the controller config to a proper Kubernetes CRD to be able to do these kinds of changes with [webhook conversion](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion).